### PR TITLE
[V3 General] Changed urban dictionary for my embed version

### PR DESCRIPTION
**Pull request by lionirdeadman**
### Type

- [ ] Bugfix
- [x] Enhancement
- [x] New feature

### Description of the changes

Replaced the old urban dictionary command to an embed version I made which adds:
- The author name 
- Thumb ups
- Thumb downs
- Reaction menu to see the different definitions

### CODE REVIEW

**Comment by lionirdeadman** [comment]
I'll update my branch (to include the changes made to V3/develop) after the code is reviewed if you don't mind. 

I don't really want to update this branch all the time while I wait for review, hope you understand.

File: redbot/cogs/general/general.py
Comment (Tobotimus) [inline redbot/cogs/general/general.py]: This help string is a bit long. Try to make the first line shorter than 72 characters, but the user should still know exactly what the command does from that line. Any further detail should be expanded after two line-breaks.

File: redbot/cogs/general/general.py
Comment (Tobotimus) [inline redbot/cogs/general/general.py]: Early declaration of variable here is unnecessary

File: redbot/cogs/general/general.py
Comment (Tobotimus) [inline redbot/cogs/general/general.py]: You'll want to return here

File: redbot/cogs/general/general.py
Comment (Tobotimus) [inline redbot/cogs/general/general.py]: Please use `"".format()` instead of concatenating strings, and also add back in the i18n call (which is surrounding the string with `_()`).

File: redbot/cogs/general/general.py
Comment (Tobotimus) [reply redbot/cogs/general/general.py]: Also, this sentence isn't quite grammatically correct, and needs to be wrapped in `_()`

File: redbot/cogs/general/general.py
Comment (Tobotimus) [inline redbot/cogs/general/general.py]: Please use `"".format()` here too

**Review by Tobotimus — CHANGES_REQUESTED** [review]
Hi, thanks for your contribution! :tada:

I like the new menu interface, much better than the old one! However there are a couple things which need to be changed in your code before we'll accept this :smiley:
 
Once you've made the changes, please post a comment to let us know.

**Comment by lionirdeadman** [comment]
I think I've changed everything you asked for. Thanks for the review. :)

Could it be in Beta 17 or is it feature-freezed? :thinking:

**Comment by lionirdeadman** [comment]
Had to come up with a fix for the 2048 character limit of descriptions, I hadn't thought of it.

Everything should be good now though.

**Comment by lionirdeadman** [comment]
Just commenting again since this is supposed to be for B18 according to the tag so I'd like a review again. Thank you.

**Comment by Kowlin** [comment]
I'm going to place this on hold to review with other core developers. See what they think.

**Comment by mikeshardmind** [comment]
This is sleek, and I like it a lot. The only problem I have with it is that it does not respect the core embed requested setting. It might be better to break this out into 2 separate functions, (old code and new) and have the appropriate one based on that setting called when the command is invoked.

**Comment by lionirdeadman** [comment]
I don't know how to respect the embed setting, if you could give me pointers, I'd be glad to do it, @mikeshardmind. 

Feel free to PM me in Discord instead of commenting here so we can do this faster! :smile:

**Comment by mikeshardmind** [comment]
There's a helper method in [context ](https://github.com/Cog-Creators/Red-DiscordBot/blob/e3bff7e87cfca23dd36bf92507a99636d7033171/redbot/core/commands/context.py#L160)for this since this particular command always should respond in the same location it was called from.

```py
if await ctx.embed_requested():
    #  output version using an embed
else:
    #  output version without an embed
```

**Comment by lionirdeadman** [comment]
I think I did it properly so this should be done now! (sorry for the wait, I was a bit busy)

**Comment by mikeshardmind** [comment]
Looks solid. :+1:

**Review by Tobotimus — APPROVED** [review]
Alright, thanks @lionirdeadman for the PR and thanks @mikeshardmind for reviewing. This has really jazzed up the urban command.

A small word of advice @lionirdeadman for any future PR's, try not to change *too much* code unnecessarily. This command has essentially been rewritten from the ground up when looking at the diff - but in retrospect it only aimed to change the output format (not any HTTP request / backend stuff). Changing more than necessary gives the reviewers more work than necessary, hence why this has taken almost 2 months to be merged! However I'll let it slide and I hope you're glad this has finally had its green button pressed :)

**Comment by lionirdeadman** [comment]
I had already made that code in a cog, I simply modified it slightly for this PR. I'll try not to do that again though if that makes things harder for you guys.

And yeah, I'm pretty happy it got merged. ^^

**Pull request merged**
